In [16]:
# Importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [17]:
# Data Loading
url = "https://archive.ics.uci.edu/ml/machine-learning-databases/secom/secom.data"
df = pd.read_csv(url, sep = ' ', header = None)
df.shape


(1567, 590)

In [18]:
df.head(10)

0        1          2          3       4      5         6       7    \
0  3030.93  2564.00  2187.7333  1411.1265  1.3602  100.0   97.6133  0.1242   
1  3095.78  2465.14  2230.4222  1463.6606  0.8294  100.0  102.3433  0.1247   
2  2932.61  2559.94  2186.4111  1698.0172  1.5102  100.0   95.4878  0.1241   
3  2988.72  2479.90  2199.0333   909.7926  1.3204  100.0  104.2367  0.1217   
4  3032.24  2502.87  2233.3667  1326.5200  1.5334  100.0  100.3967  0.1235   
5  2946.25  2432.84  2233.3667  1326.5200  1.5334  100.0  100.3967  0.1235   
6  3030.27  2430.12  2230.4222  1463.6606  0.8294  100.0  102.3433  0.1247   
7  3058.88  2690.15  2248.9000  1004.4692  0.7884  100.0  106.2400  0.1185   
8  2967.68  2600.47  2248.9000  1004.4692  0.7884  100.0  106.2400  0.1185   
9  3016.11  2428.37  2248.9000  1004.4692  0.7884  100.0  106.2400  0.1185   

      8       9    ...     580       581     582     583     584      585  \
0  1.5005  0.0162  ...     NaN       NaN  0.5005  0.0118  0.0035   2.3630   
1  1.4966 -0.0005  ...  0.0060  208.2045  0.5019  0.0223  0.0055   4.4447   
2  1.4436  0.0041  ...  0.0148   82.8602  0.4958  0.0157  0.0039   3.1745   
3  1.4882 -0.0124  ...  0.0044   73.8432  0.4990  0.0103  0.0025   2.0544   
4  1.5031 -0.0031  ...     NaN       NaN  0.4800  0.4766  0.1045  99.3032   
5  1.5287  0.0167  ...  0.0052   44.0077  0.4949  0.0189  0.0044   3.8276   
6  1.5816 -0.0270  ...     NaN       NaN  0.5010  0.0143  0.0042   2.8515   
7  1.5153  0.0157  ...  0.0063   95.0310  0.4984  0.0106  0.0034   2.1261   
8  1.5358  0.0111  ...  0.0045  111.6525  0.4993  0.0172  0.0046   3.4456   
9  1.5381  0.0159  ...  0.0073   90.2294  0.4967  0.0152  0.0038   3.0687   

      586     587     588       589  
0     NaN     NaN     NaN       NaN  
1  0.0096  0.0201  0.0060  208.2045  
2  0.0584  0.0484  0.0148   82.8602  
3  0.0202  0.0149  0.0044   73.8432  
4  0.0202  0.0149  0.0044   73.8432  
5  0.0342  0.0151  0.0052   44.0077  
6  0.0342  0.0151  0.0052   44.0077  
7  0.0204  0.0194  0.0063   95.0310  
8  0.0111  0.0124  0.0045  111.6525  
9  0.0212  0.0191  0.0073   90.2294  

[10 rows x 590 columns]

In [8]:
df.tail(10)

0        1          2          3       4      5         6       7    \
1557  3072.10  2534.87  2177.4333  2945.8855  1.3321  100.0   83.1700  0.1253   
1558  3012.30  2466.84  2217.4111  1032.2836  1.4802  100.0  101.3511  0.1195   
1559  3076.33  2456.13  2217.4111  1032.2836  1.4802  100.0  101.3511  0.1195   
1560  2770.40  2549.42  2204.2889  2637.9989  1.5549  100.0   86.1089  0.1234   
1561  2951.14  2326.59  2212.6334  1081.5662  1.0096  100.0  113.4278  0.1253   
1562  2899.41  2464.36  2179.7333  3085.3781  1.4843  100.0   82.2467  0.1248   
1563  3052.31  2522.55  2198.5667  1124.6595  0.8763  100.0   98.4689  0.1205   
1564  2978.81  2379.78  2206.3000  1110.4967  0.8236  100.0   99.4122  0.1208   
1565  2894.92  2532.01  2177.0333  1183.7287  1.5726  100.0   98.7978  0.1213   
1566  2944.92  2450.76  2195.4444  2914.1792  1.5978  100.0   85.1011  0.1235   

         8       9    ...     580       581     582     583     584     585  \
1557  1.4359 -0.0114  ...  0.0023   26.7330  0.5010  0.0132  0.0035  2.6249   
1558  1.3832  0.0042  ...  0.0054  176.6783  0.4993  0.0130  0.0037  2.5976   
1559  1.3120 -0.0043  ...  0.0057  117.4564  0.4960  0.0157  0.0036  3.1743   
1560  1.2811  0.0037  ...  0.0042  127.3154  0.5009  0.0155  0.0036  3.0997   
1561  1.4492 -0.0134  ...  0.0051   46.4573  0.4965  0.0118  0.0032  2.3817   
1562  1.3424 -0.0045  ...  0.0047  203.1720  0.4988  0.0143  0.0039  2.8669   
1563  1.4333 -0.0061  ...     NaN       NaN  0.4975  0.0131  0.0036  2.6238   
1564     NaN     NaN  ...  0.0025   43.5231  0.4987  0.0153  0.0041  3.0590   
1565  1.4622 -0.0072  ...  0.0075   93.4941  0.5004  0.0178  0.0038  3.5662   
1566     NaN     NaN  ...  0.0045  137.7844  0.4987  0.0181  0.0040  3.6275   

         586     587     588       589  
1557  0.0277  0.0074  0.0023   26.7330  
1558  0.0097  0.0172  0.0054  176.6783  
1559  0.0150  0.0176  0.0057  117.4564  
1560  0.0105  0.0133  0.0042  127.3154  
1561  0.0320  0.0148  0.0051   46.4573  
1562  0.0068  0.0138  0.0047  203.1720  
1563  0.0068  0.0138  0.0047  203.1720  
1564  0.0197  0.0086  0.0025   43.5231  
1565  0.0262  0.0245  0.0075   93.4941  
1566  0.0117  0.0162  0.0045  137.7844  

[10 rows x 590 columns]

590 sensor signals collected from the wafer fabrication line.
1567 rows (wafers being manufactured) and 590 columns (sensor readings).

In [ ]:
# Save the data to csv
data_path = os.mkdir()

EXPLANATORY DATA ANALYSIS

In [15]:
# Data quality check
df.isnull().sum()

0       6
1       7
2      14
3      14
4      14
       ..
585     1
586     1
587     1
588     1
589     1
Length: 590, dtype: int64

We are assuming that the Nan values are simply attributed to sensors that were off or those that barely ever fired.